### PR TITLE
Add definition for /activity endpoint

### PR DESCRIPTION
**Pull request by plamut**
Towards https://github.com/tradingstrategy-ai/backend/issues/65

### CODE REVIEW

**Comment by kenkunz** [comment]
### Two suggestions
1. use a JSON payload instead of query param – this provides more flexibility if we later wish to add additional payload data.
2. name the param `user_id` instead of `unique_id` – more semantic so we know what the _uniqueness_ is based on.

### Suggested payload
#### `POST /chart-activity`
```json
{
  "user_id": "LN5OQsPfTb2sXZYVWatamw"
}
```

`user_id` = URL-safe base64-encoded UUID (v4) – 22 characters in range `[a-zA-Z_-]`

**Comment by kenkunz** [comment]
@miohtama, @plamut …

### One additional suggestion

We could make the endpoint more generic (e.g., `/activity`) and include an activity type in the payload. If we later find a need to track some other type of activity from `frontend`, the `backend` API would support this with no changes.

I'm guessing the LOE would be similar either way.

### Suggested payload
#### `POST /activity`
```json
{
  "user_id": "LN5OQsPfTb2sXZYVWatamw",
  "type": "chart"
}
```

`user_id` = URL-safe base64-encoded UUID (v4) – 22 characters in range `[a-zA-Z_-]`
`type` = a string representing the type of activity being tracked

**Comment by plamut** [comment]
>     1. use a JSON payload instead of query param – this provides more flexibility if we later wish to add additional payload data.

On the backend side it doesn't really matter much in terms of flexibility. The API spec would still have to be updated on any change, while implementation-wise it would be roughly the same. If we were to accept a generic JSON payload, params would, generally, have to be extracted manually, but it's easy to do it even by hand.

Would the frontend benefit significantly by having a JSON payload?


 
>     2. name the param `user_id` instead of `unique_id` – more semantic so we know what the _uniqueness_ is based on.

Sounds reasonable, let's rename it. :+1: 

> We could make the endpoint more generic (e.g., `/activity`) and include an activity type in the payload. 

I'm fine with that as well. :+1: Easier to add new activity types than defining additional endpoints.

@miohtama What are your thoughts?

IMO, since this endpoint is for internal purposes, we have much more flexibilty in (un)designing the schema and changing it at will, so I wouldn't worry to much about "getting it right" right off the bat, as long as it serves its purpose (collecting chart interaction data).

**Comment by kenkunz** [comment]
> IMO, since this endpoint is for internal purposes, we have much more flexibility in (un)designing the schema and changing it at will, so I wouldn't worry to much about "getting it right" right off the bat, as long as it serves its purpose (collecting chart interaction data).

👍 makes sense … I concur!

> On the backend side it doesn't really matter much in terms of flexibility. The API spec would still have to be updated on any change, while implementation-wise it would be roughly the same. If we were to accept a generic JSON payload, params would, generally, have to be extracted manually, but it's easy to do it even by hand.

I was thinking more about the flexibility of the API spec itself to accommodate changes. GET params are fine for simple, flat data, but they GET ugly quickly (😉) if you have a lot of params or nesting. If we start with JSON we're just adjusting the JSON (hence more flexible).

To your other point above, though … it doesn't matter that much right now and it's easy to change later, so I don't really have a strong opinion.

> Would the frontend benefit significantly by having a JSON payload?

No difference with our current payload. JSON payload would be simpler if we end up with more complex payload in the future.

**Bottom line:** go with what feels best to you at this point, we can adjust later.

**Comment by plamut** [comment]
I'll go with JSON then and see how it goes. If I hit any unexpected trouble during the implementation, we can re-discuss.

I will update this PR once the implementation PR is ready, and when it proves that the chosen approach does not complicate the implementation unnecessarily.

**Comment by plamut** [comment]
Updated the endpoint definition based on the implementation in https://github.com/tradingstrategy-ai/backend/pull/69

PTAL

(we need this to update the `spec` in the backend repo to unblock the [main PR](https://github.com/tradingstrategy-ai/backend/pull/69))

**Comment by plamut** [comment]
The design spec changes have been approved in the [implementation PR](https://github.com/tradingstrategy-ai/backend/pull/69), meaning we can merge this.
